oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/internal/OIDCUserManager.java

```diff
@@ -304,10 +308,129 @@ private Principal updateUser(IDTokenClaimsSet idToken, UserInfo userInfo) throws
         return new SimplePrincipal(userDocument.getPrefixedFullName());
     }
 
+    /**
+     * Remove user name from provided XWiki group.
+     *
+     * @param xwikiUserName the full name of the user.
+     * @param groupName the name of the group.
+     * @param context the XWiki context.
+     */
+    protected void removeUserFromXWikiGroup(String xwikiUserName, String groupName, XWikiContext context)
+    {
+        try {
+            BaseClass groupClass = context.getWiki().getGroupClass(context);
+
+            // Get the XWiki document holding the objects comprising the group membership list
+            XWikiDocument groupDoc = context.getWiki().getDocument(groupName, context);
+
+            synchronized (groupDoc) {
+                // Get and remove the specific group membership object for the user
+                BaseObject groupObj =
+                        groupDoc.getXObject(groupClass.getDocumentReference(), XWIKI_GROUP_MEMBERFIELD, xwikiUserName);
+                groupDoc.removeXObject(groupObj);
+
+                // Save modifications
+                context.getWiki().saveDocument(groupDoc, context);
+            }
+        } catch (Exception e) {
+            this.logger.error("Failed to remove a user from a group " + xwikiUserName + " group: " + groupName, e);
+        }
+    }
+
+    protected void addUserToXWikiGroup(String xwikiUserName, String groupName, XWikiContext context)
+    {
+        try {
+            BaseClass groupClass = context.getWiki().getGroupClass(context);
+
+            // Get document representing group
+            XWikiDocument groupDoc = context.getWiki().getDocument("XWiki." + groupName, context);
+
+            this.logger.info("Adding user [{}] to xwiki group [{}]. groupClass: [{}]; groupDoc: [{}]", xwikiUserName, groupName, groupClass, groupDoc);
```

Comment:
I would make this a debug log, I don't think it's right to fill the general log with this kind of information.